File: 2022/day2/elixir/part1.exs
defmodule MyModule do
  @rock "X"
  @paper "Y"
  @scissors "Z"
  def main do
    m = %{
      ["A", @rock] => 1 + 3,
      ["B", @rock] => 1 + 0,
      ["C", @rock] => 1 + 6,
      ["A", @paper] => 2 + 6,
      ["B", @paper] => 2 + 3,
      ["C", @paper] => 2 + 0,
      ["A", @scissors] => 3 + 0,
      ["B", @scissors] => 3 + 6,
      ["C", @scissors] => 3 + 3
    }

    contents =
      File.read!("../input.txt") |> String.split("\r\n") |> Enum.filter(fn line -> line != "" end)

    total =
      Enum.reduce(contents, 0, fn line, acc ->
        [opponent, player] = String.split(line, " ")
        result = Map.get(m, [opponent, player])
        acc + result
      end)

    IO.puts(total)
  end
end

MyModule.main()

File: 2022/day2/cpp/part1.cpp
#include <fstream>
#include <iostream>

using namespace std;

int main() {
    ifstream f("../input.txt");
    char x, z;
    int score{};
    while (f >> x) {
        f >> z;
        if (z == 'X') {
            ++score;
            if (x == 'A')
                score += 3;
            if (x == 'C')
                score += 6;
        } else if (z == 'Y') {
            score += 2;
            if (x == 'A')
                score += 6;
            if (x == 'B')
                score += 3;
        } else if (z == 'Z') {
            score += 3;
            if (x == 'B')
                score += 6;
            if (x == 'C')
                score += 3;
        }
    }
    cout << score << endl;
}
